common/src/main/java/org/apache/nemo/common/Util.java

```diff
@@ -117,7 +117,10 @@ public static IREdge cloneEdge(final IREdge edgeToClone,
                                  final IRVertex newSrc,
                                  final IRVertex newDst) {
     return cloneEdge(
-      edgeToClone.getPropertyValue(CommunicationPatternProperty.class).get(), edgeToClone, newSrc, newDst);
+      edgeToClone.getPropertyValue(CommunicationPatternProperty.class).orElseThrow(IllegalStateException::new),
```

Comment:
I'm seeing a LOT of these lines. I'm just wondering whether it would be simply enough to return `.orElse(null)`, if the property is empty, instead of throwing an exception. What do you think?

Reply:
IMHO, it is not good to propagate the error with null. If we just return null if the `Optional` is empty, we can't know where the "nulled' value occurs NPE.

Reply:
Actually, this wasn't an appropriate place to mention the problem. I think this case would be happier if it was left as-is.